src/components/FilterView.js

```diff
@@ -0,0 +1,119 @@
+import { useState } from "react";
+
+
+const idealFor  = [ 'Boys', 'Girls', 'Men', 'Unisex', 'Women' ];
+const size      = ['S', 'M' , 'L' , 'XXL', 'XXXL'];
+const topBrands = ['IMARA', 'Biba' , 'Libas' , 'Manyavar', 'KISAH'];
+
+
+const FILTER_BRAND     = 'brand';
+const FILTER_SIZE      = 'size';
+const FILTER_IDEAL_FOR = 'ideal_for';
+const FILTER_MIN_PRICE = 'minPrice';
+const FILTER_MAX_PRICE = 'maxPrice';
+
+function FilterView({updateFilters}) {
+
+  const initialFilterState = {};
+  initialFilterState[FILTER_BRAND] = new Set([]);
+  initialFilterState[FILTER_IDEAL_FOR] = new Set([]);
+  initialFilterState[FILTER_SIZE] = new Set([]);
+  initialFilterState[FILTER_MIN_PRICE] = 0;
+  initialFilterState[FILTER_MAX_PRICE] = 1000;
+
+  const [filters, setFilters] = useState(initialFilterState);
+
+  function onFilterChange(e, filter, v){
```

Comment:
@rahulmeghwal it seems like you could reduce the required parameters and just pass the event since you can extract the value from the event (vs passing the redundant value param). Maybe there's something this extra param can support though.

Reply:
I used the extra value(v) param to get the corresponding label value of the checked box, for example if checkbox for the brand 'Biba' is checked then 'v' will have value 'Biba', the event only tells me if the checkbox was checked or not. I may not be aware of other ways the same can be achieved more efficiently. Also I have removed the 'Apply Filters' button and now products are fetched as soon as one changes the filters i.e. no need to click Apply Filters button.